packages/mdc-list/_mixins.scss

```diff
@@ -21,62 +21,397 @@
+  $divider-color: if(
+    mdc-theme-tone($mdc-theme-background) == "dark",
+    $mdc-list-divider-color-on-dark-bg,
+    $mdc-list-divider-color-on-light-bg
+  );
+
+  @include mdc-ripple-common($query);
+
+  .mdc-list {
+    @include mdc-list-base_($query);
+  }
+
+  @include mdc-list-item-secondary-text-ink-color(text-secondary-on-background, $query);
+  @include mdc-list-item-graphic-fill-color(transparent, $query);
+  @include mdc-list-item-graphic-ink-color(text-icon-on-background, $query);
+  @include mdc-list-item-meta-ink-color(text-hint-on-background, $query);
+  @include mdc-list-group-subheader-ink-color(text-primary-on-background, $query);
+
+  .mdc-list--dense {
+    @include mdc-feature-targets($feat-structure) {
+      padding-top: 4px;
+      padding-bottom: 4px;
+      font-size: .812rem;
+    }
+  }
+
+  .mdc-list-item {
+    @include mdc-feature-targets($feat-structure) {
+      @include mdc-list-item-base_;
+    }
+  }
+
+  // "Selected" is ephemeral and likely to change soon. E.g., selecting one or more photos to share in Google Photos.
+  // "Activated" is more permanent. E.g., the currently highlighted navigation destination in a drawer.
+  .mdc-list-item--selected,
+  .mdc-list-item--activated {
+    @include mdc-list-item-primary-text-ink-color(primary, $query);
+    @include mdc-list-item-graphic-ink-color(primary, $query);
+  }
+
+  .mdc-list-item--disabled {
+    @include mdc-list-item-primary-text-ink-color(text-disabled-on-background, $query);
+  }
+
+  .mdc-list-item__graphic {
+    @include mdc-feature-targets($feat-structure) {
+      @include mdc-list-graphic-size_(24px);
+
+      flex-shrink: 0;
+      align-items: center;
+      justify-content: center;
+      fill: currentColor;
+    }
+  }
+
+  // Extra specificity is to override .material-icons display style if used in
+  // conjunction with mdc-list-item__graphic
+  // stylelint-disable plugin/selector-bem-pattern
+  .mdc-list .mdc-list-item__graphic {
+    @include mdc-feature-targets($feat-structure) {
+      display: inline-flex;
+    }
+  }
+  // stylelint-enable plugin/selector-bem-pattern
+
+  .mdc-list-item__meta {
+    @include mdc-feature-targets($feat-structure) {
+      @include mdc-rtl-reflexive-property(margin, auto, 0, ".mdc-list-item");
+    }
+  }
+
+  .mdc-list-item__text {
+    @include mdc-typography-overflow-ellipsis($query);
+  }
+
+  // Disable interaction on label elements that may automatically
+  // toggle corresponding checkbox / radio input.
+  .mdc-list-item__text[for] {
+    @include mdc-feature-targets($feat-structure) {
+      pointer-events: none;
+    }
+  }
+
+  .mdc-list-item__primary-text {
+    @include mdc-typography-overflow-ellipsis($query);
+    @include mdc-typography-baseline-top($item-primary-text-baseline-height, $query);
+    @include mdc-typography-baseline-bottom($item-secondary-text-baseline-height, $query);
+    @include mdc-typography-baseline-top($dense-item-primary-text-baseline-height, $query);
```

Comment:
This line is in the wrong place. This belongs inside `.mdc-list--dense & {` below.